packages/doenetml-worker-javascript/src/utils/debug/propType.ts

```diff
@@ -0,0 +1,151 @@
+import { createComponentInfoObjects } from "../componentInfoObjects";
+
+const componentInfoObjects = createComponentInfoObjects();
+
+/**
+ * Attempt to determine the component type of `propName` from `componentType`.
+ *
+ * If successful, returns the component type. Throws an error if it fails.
+ *
+ * Function will fail if `propName` is not a public state variable of `componentType`.
+ *
+ * Function will also fail for the edge case where `propName` has variable component type
+ */
+export function determinePropType(
+    componentType: string,
+    propName: string,
+    nIndices = 0,
```

Comment:
This is the number of times it is indexed? I.e., for `$point.x[1][1]`, `nIndices === 2`?

Reply:
That's right.